sdk/cosmos/azure-cosmos/azure/cosmos/_cosmos_client_connection.py

```diff
@@ -24,48 +24,55 @@
 
 """Document client class for the Azure Cosmos database service.
 """
-# https://github.com/PyCQA/pylint/issues/3112
-# Currently pylint is locked to 2.3.3 and this is fixed in 2.4.4
-from typing import Dict, Any, Optional, TypeVar  # pylint: disable=unused-import
+import re
+from typing import Callable, Dict, Any, Iterable, List, Mapping, Optional, Sequence, Tuple, Union, cast
 import urllib.parse
+
 from urllib3.util.retry import Retry
-from azure.core.paging import ItemPaged  # type: ignore
-from azure.core import PipelineClient  # type: ignore
-from azure.core.pipeline.policies import (  # type: ignore
+from azure.core.credentials import TokenCredential
+from azure.core.paging import ItemPaged
+from azure.core import PipelineClient
+from azure.core.pipeline.policies import (
     HTTPPolicy,
     ContentDecodePolicy,
     HeadersPolicy,
     UserAgentPolicy,
     NetworkTraceLoggingPolicy,
     CustomHookPolicy,
     DistributedTracingPolicy,
-    ProxyPolicy)
+    ProxyPolicy
+)
 
 from . import _base as base
 from . import documents
-from .documents import ConnectionPolicy
+from .documents import ConnectionPolicy, DatabaseAccount
 from ._constants import _Constants as Constants
 from . import http_constants, exceptions
 from . import _query_iterable as query_iterable
 from . import _runtime_constants as runtime_constants
-from . import _request_object
+from ._request_object import RequestObject
 from . import _synchronized_request as synchronized_request
 from . import _global_endpoint_manager as global_endpoint_manager
 from ._routing import routing_map_provider, routing_range
 from ._retry_utility import ConnectionRetryPolicy
-from . import _session
+from ._session import Session
 from . import _utils
 from .partition_key import _Undefined, _Empty, PartitionKey
 from ._auth_policy import CosmosBearerTokenCredentialPolicy
 from ._cosmos_http_logging_policy import CosmosHttpLoggingPolicy
+from ._range_partition_resolver import RangePartitionResolver
 
-ClassType = TypeVar("ClassType")
 
+# Cosmos table validation regex breakdown:
+# ^ Match start of string.
+# [^/\#?]{0,254} Match any character that is not /\#? for between 0-254 characters.
+# $ End of string
+_VALID_COSMOS_RESOURCE = re.compile(r"^[^/\\#?\t\r\n]{0,254}$")
```

Comment:
This should be tested to confirm the correct parameters for a valid cosmos resource.

Reply:
so long as this passes the tests we have for resource id validation we should be good